Util/Xamarin.Build.Download/build.cake

```diff
@@ -1,28 +1,54 @@
-#load "../../common.cake"
+var TARGET = Argument ("t", Argument ("target", "ci"));
 
-var TARGET = Argument ("t", Argument ("target", "Default"));
+var SOURCE_COMMIT = EnvironmentVariable("BUILD_SOURCEVERSION") ?? "";
+var SOURCE_BRANCH = EnvironmentVariable("BUIlD_SOURCEBRANCHNAME") ?? "";
 
-var buildSpec = new BuildSpec {
-	Libs = new [] { 
-		new DefaultSolutionBuilder {
-			SolutionPath = "source/Xamarin.Build.Download.sln",
-			BuildsOn = BuildPlatforms.Windows | BuildPlatforms.Mac,
-			OutputFiles = new [] { 
-				new OutputFileCopy { FromFile = "./source/Xamarin.Build.Download/bin/Release/Xamarin.Build.Download.dll" },
-				new OutputFileCopy { FromFile = "./source/Xamarin.Build.Download/bin/Release/Xamarin.Build.Download.targets" },
-				new OutputFileCopy { FromFile = "./source/Xamarin.Build.Download/bin/Release/Xamarin.Build.Download.props" },
-				new OutputFileCopy { FromFile = "./source/Xamarin.Build.Download/bin/Release/System.Net.Http.Formatting.dll" },
-				new OutputFileCopy { FromFile = "./source/Xamarin.Build.Download/bin/Release/Newtonsoft.Json.dll" },
-				new OutputFileCopy { FromFile = "./source/Xamarin.Build.Download/bin/Release/Mono.Cecil.dll" },
-			},
+Task("libs")
+	.Does(() =>
+{
+	MSBuild("./source/Xamarin.Build.Download.sln", c => {
+		c.Configuration = "Release";
+		c.Restore = true;
+	});
+});
 
-		}
-	},
-	NuGets = new [] {
-		new NuGetInfo { NuSpec = "./nuget/Xamarin.Build.Download.nuspec", RequireLicenseAcceptance = true },
-	}
-};
+Task("nuget")
+	.IsDependentOn("libs")
+	.Does(() =>
+{
+	MSBuild ("./source/Xamarin.Build.Download.sln", c => {
+		c.Configuration = "Release";
+		c.Restore = true;
+		c.Targets.Clear();
+		c.Targets.Add("Pack");
+		c.Properties.Add("PackageOutputPath", new [] { MakeAbsolute(new FilePath("./output")).FullPath });
+		if (!string.IsNullOrEmpty(SOURCE_BRANCH))
+			c.Properties.Add("RepositoryBranch", new [] { SOURCE_BRANCH });
+		if (!string.IsNullOrEmpty(SOURCE_COMMIT))
+			c.Properties.Add("RepositoryCommit", new [] { SOURCE_COMMIT });
```

Comment:
Since we may actually want to force this on all builds, we could do this in the template:

https://github.com/xamarin/XamarinComponents/blob/master/.ci/build.yml#L187-L188